src/clj/rems/db/licenses.clj

```diff
@@ -1,79 +1,115 @@
 (ns rems.db.licenses
-  "querying localized licenses"
-  (:require [medley.core :refer [distinct-by]]
-            [rems.common.util :refer [apply-filters]]
-            [rems.db.core :as db]))
-
-(defn- format-license [license]
-  {:id (:id license)
-   :licensetype (:type license)
-   :organization (:organization license)
-   :enabled (:enabled license)
-   :archived (:archived license)})
-
-(defn- format-licenses [licenses]
-  (mapv format-license licenses))
-
-(defn- get-license-localizations []
-  (->> (db/get-license-localizations)
-       (map #(update-in % [:langcode] keyword))
-       (group-by :licid)))
-
-(defn- localize-license [localizations license]
-  (assoc license :localizations
-         (into {} (for [{:keys [langcode title textcontent attachmentid]} (get localizations (:id license))]
-                    [langcode {:title title
-                               :textcontent textcontent
-                               :attachment-id attachmentid}]))))
-
-(defn- localize-licenses [licenses]
-  (mapv (partial localize-license (get-license-localizations)) licenses))
-
-(defn get-resource-licenses
-  "Get resource licenses for given resource id"
-  [id]
-  (->> (db/get-resource-licenses {:id id})
-       (format-licenses)
-       (localize-licenses)))
+  (:require [clojure.set]
+            [medley.core :refer [assoc-some map-vals]]
+            [rems.cache :as cache]
+            [rems.common.util :refer [apply-filters index-by]]
+            [rems.db.core :as db]
+            [schema.coerce :as coerce]
+            [schema.core :as s]))
+
+(s/defschema LicenseDb
+  {:id s/Int
+   :licensetype (s/enum "link" "text" "attachment")
+   :organization s/Str
+   :enabled s/Bool
+   :archived s/Bool})
+
+(def ^:private coerce-LicenseDb
+  (coerce/coercer! LicenseDb coerce/string-coercion-matcher))
+
+(defn- parse-license-raw [x]
+  (let [license {:id (:id x)
+                 :licensetype (:type x)
+                 :organization (:organization x)
+                 :enabled (:enabled x)
+                 :archived (:archived x)}]
+    (coerce-LicenseDb license)))
```

Comment:
alternative namings (raw is maybe unnecessary?):
- `parse-license-from-db`
- `parse-license-db`